plugin/src/main/java/org/opensearch/ml/action/agents/GetAgentTransportAction.java

```diff
@@ -42,74 +51,106 @@
+        String tenantId = mlAgentGetRequest.getTenantId();
         boolean isSuperAdmin = isSuperAdminUserWrapper(clusterService, client);
 
-        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
-            client.get(getRequest, ActionListener.runBefore(ActionListener.wrap(r -> {
-                log.debug("Completed Get Agent Request, id:{}", agentId);
+        FetchSourceContext fetchSourceContext = new FetchSourceContext(true, Strings.EMPTY_ARRAY, Strings.EMPTY_ARRAY);
+        GetDataObjectRequest getDataObjectRequest = new GetDataObjectRequest.Builder()
+            .index(ML_AGENT_INDEX)
+            .id(agentId)
+            .tenantId(tenantId)
+            .fetchSourceContext(fetchSourceContext)
+            .build();
 
-                if (r != null && r.isExists()) {
-                    try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, r.getSourceAsBytesRef())) {
-                        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
-                        MLAgent mlAgent = MLAgent.parse(parser);
-                        if (mlAgent.getIsHidden() && !isSuperAdmin) {
-                            actionListener
-                                .onFailure(
-                                    new OpenSearchStatusException(
-                                        "User doesn't have privilege to perform this operation on this agent",
-                                        RestStatus.FORBIDDEN
-                                    )
-                                );
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            sdkClient
+                .getDataObjectAsync(getDataObjectRequest, client.threadPool().executor(GENERAL_THREAD_POOL))
+                .whenComplete((r, throwable) -> {
+                    context.restore();
+                    log.debug("Completed Get Agent Request, Agent id:{}", agentId);
+                    if (throwable != null) {
+                        Exception cause = SdkClientUtils.unwrapAndConvertToException(throwable);
+                        if (cause instanceof IndexNotFoundException) {
+                            log.error("Failed to get Agent index", cause);
+                            actionListener.onFailure(new OpenSearchStatusException("Failed to get agent index", RestStatus.NOT_FOUND));
                         } else {
-                            actionListener.onResponse(MLAgentGetResponse.builder().mlAgent(mlAgent).build());
+                            log.error("Failed to get ML Agent {}", agentId, cause);
+                            actionListener.onFailure(cause);
+                        }
+                    } else {
+                        try {
+                            GetResponse gr = GetResponse.fromXContent(r.parser());
+                            if (gr.isExists()) {
```

Comment:
Exception handling and parsing `GetResponse` has duplicate code across `DeleteAgentTransportAction` and `FetchAgentTransportAction` class. Move it a to common utility.

Reply:
> Move it a to common utility.

Not sure what is going to be most common here, as there's a lot of variation inside these blocks.

It's also much easier for this initial migration for review to literally copy/paste the existing (duplicate) contents.

Reply:
Yea, not only for agents, I saw there any other apis where get functionalities could be put into a common utility. I was thinking to address those in separate PRs.